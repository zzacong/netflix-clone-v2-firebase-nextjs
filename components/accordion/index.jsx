import { useState, useContext, createContext } from 'react'
import { MdAdd, MdClose } from 'react-icons/md'
import {
  Container,
  Frame,
  Title,
  Item,
  Inner,
  Header,
  Body,
} from './styles/accordion'

const ToggleContext = createContext()

export default function Accordion({ children, ...props }) {
  return (
    <Container {...props}>
      <Inner>{children}</Inner>
    </Container>
  )
}

Accordion.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Accordion.Frame = ({ children, ...props }) => {
  return <Frame {...props}>{children}</Frame>
}

Accordion.Item = ({ children, ...props }) => {
  const [toggleShow, setToggleShow] = useState(false)

  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...props}>{children}</Item>
    </ToggleContext.Provider>
  )
}

Accordion.Header = ({ children, ...props }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext)

  return (
    <Header onClick={() => setToggleShow(prev => !prev)} {...props}>
      {children}
      {toggleShow ? <MdClose /> : <MdAdd />}
    </Header>
  )
}

Accordion.Body = ({ children, ...props }) => {
  const { toggleShow } = useContext(ToggleContext)

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...props}>
      <span>{children}</span>
    </Body>
  )
}
