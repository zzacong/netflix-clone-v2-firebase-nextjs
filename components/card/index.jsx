import React, { useState, useContext, createContext } from 'react'
import capitalize from 'lodash.capitalize'
import { MdClose } from 'react-icons/md'
import {
  Container,
  Group,
  Title,
  Subtitle,
  Text,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Maturity,
  Content,
  Meta,
  Entities,
  Item,
  Image,
} from './styles/card'

export const FeatureContext = createContext()

export default function Card({ children, ...props }) {
  const [showFeature, setShowFeature] = useState(false)
  const [itemFeature, setItemFeature] = useState({})

  return (
    <FeatureContext.Provider
      value={{ showFeature, setShowFeature, itemFeature, setItemFeature }}
    >
      <Container {...props}>{children}</Container>
    </FeatureContext.Provider>
  )
}

Card.Group = ({ children, ...props }) => <Group {...props}>{children}</Group>

Card.Title = ({ children, ...props }) => <Title {...props}>{children}</Title>

Card.Subtitle = ({ children, ...props }) => (
  <Subtitle {...props}>{children}</Subtitle>
)

Card.Text = ({ children, ...props }) => <Text {...props}>{children}</Text>

Card.Entities = ({ children, ...props }) => (
  <Entities {...props}>{children}</Entities>
)

Card.Meta = ({ children, ...props }) => <Meta {...props}>{children}</Meta>

Card.Item = ({ item, children, ...props }) => {
  const { setShowFeature, setItemFeature } = useContext(FeatureContext)

  return (
    <Item
      onClick={() => {
        setItemFeature(item)
        setShowFeature(true)
      }}
      {...props}
    >
      {children}
    </Item>
  )
}

Card.Image = ({ ...props }) => <Image {...props} />

Card.Feature = function CardFeature({ children, category, ...props }) {
  const { showFeature, itemFeature, setShowFeature } = useContext(
    FeatureContext
  )

  if (!showFeature) return null

  return (
    <Feature
      {...props}
      src={`/images/${category}/${itemFeature.genre}/${itemFeature.slug}/large.jpg`}
    >
      <Content>
        <FeatureTitle>{itemFeature.title}</FeatureTitle>
        <FeatureText>{itemFeature.description}</FeatureText>
        <FeatureClose onClick={() => setShowFeature(false)}>
          <MdClose />
        </FeatureClose>

        <Group margin="30px 0" flexDirection="row" alignItems="center">
          <Maturity rating={itemFeature.maturity}>
            {itemFeature.maturity < 12 ? 'PG' : itemFeature.maturity}
          </Maturity>
          <FeatureText fontWeight="bold">
            {capitalize(itemFeature.genre)}
          </FeatureText>
        </Group>

        {children}
      </Content>
    </Feature>
  )
}
