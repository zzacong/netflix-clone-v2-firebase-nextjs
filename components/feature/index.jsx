import { Container, Title, SubTitle } from './styles/feature'

export default function Feature({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

Feature.Title = ({ children, ...props }) => <Title {...props}>{children}</Title>

Feature.SubTitle = ({ children, ...props }) => (
  <SubTitle {...props}>{children}</SubTitle>
)
