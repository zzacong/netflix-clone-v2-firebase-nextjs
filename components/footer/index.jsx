import {
  Container,
  Row,
  Column,
  Link,
  Title,
  Text,
  Break,
} from './styles/footer'

export default function Footer({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

Footer.Row = ({ children, ...props }) => {
  return <Row {...props}>{children}</Row>
}

Footer.Column = ({ children, ...props }) => {
  return <Column {...props}>{children}</Column>
}

Footer.Link = ({ children, ...props }) => {
  return <Link {...props}>{children}</Link>
}

Footer.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}

Footer.Text = ({ children, ...props }) => {
  return <Text {...props}>{children}</Text>
}

Footer.Break = ({ ...restProps }) => {
  return <Break {...restProps} />
}
