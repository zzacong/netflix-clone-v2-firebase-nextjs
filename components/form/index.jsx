import { forwardRef } from 'react'
import NextLink from 'next/link'
import {
  Container,
  Error,
  Base,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form'

export default function Form({ children, ...props }) {
  return <Container {...props}>{children}</Container>
}

Form.Base = ({ children, ...props }) => <Base {...props}>{children}</Base>

Form.Error = ({ children, ...props }) => <Error {...props}>{children}</Error>

Form.Title = ({ children, ...props }) => <Title {...props}>{children}</Title>

Form.Text = ({ children, ...props }) => <Text {...props}>{children}</Text>

Form.TextSmall = ({ children, ...props }) => (
  <TextSmall {...props}>{children}</TextSmall>
)

Form.Link = ({ href, children, ...props }) => (
  <NextLink href={href} passHref>
    <Link {...props}>{children}</Link>
  </NextLink>
)

Form.Input = forwardRef(({ children, ...props }, ref) => (
  <Input {...props} ref={ref}>
    {children}
  </Input>
))

Form.Submit = ({ children, ...props }) => <Submit {...props}>{children}</Submit>
