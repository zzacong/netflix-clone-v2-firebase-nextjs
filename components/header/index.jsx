import { useState } from 'react'
import NextLink from 'next/link'
import { MdInfoOutline, MdSearch } from 'react-icons/md'
import { IoMdPlay } from 'react-icons/io'
import {
  Container,
  Group,
  Background,
  Dropdown,
  Picture,
  Link,
  Search,
  Profile,
  FeatureCallOut,
  SearchIcon,
  SearchInput,
  ButtonLink,
  PlayButton,
  InfoButton,
  Text,
  Feature,
  Logo,
} from './styles/header'

export default function Header({ bg = true, children, ...props }) {
  return bg ? <Background {...props}>{children}</Background> : children
}

Header.Frame = ({ children, ...props }) => (
  <Container {...props}>{children}</Container>
)

Header.Group = ({ children, ...props }) => <Group {...props}>{children}</Group>

Header.Logo = ({ href, ...props }) => (
  <NextLink href={href} passHref>
    <a>
      <Logo {...props} />
    </a>
  </NextLink>
)

Header.ButtonLink = ({ href, children, ...props }) => (
  <NextLink href={href} passHref>
    <ButtonLink {...props}>{children}</ButtonLink>
  </NextLink>
)

Header.Feature = ({ children, ...props }) => (
  <Feature {...props}>{children}</Feature>
)

Header.FeatureCallOut = ({ children, ...props }) => (
  <FeatureCallOut {...props}>{children}</FeatureCallOut>
)

Header.Text = ({ children, ...props }) => <Text {...props}>{children}</Text>

Header.TextLink = ({ children, ...props }) => <Link {...props}>{children}</Link>

Header.Profile = ({ children, ...props }) => (
  <Profile {...props}>{children}</Profile>
)

Header.Picture = ({ src, ...props }) => (
  <Picture src={`/images/users/${src}.png`} {...props} />
)

Header.Dropdown = ({ children, ...props }) => (
  <Dropdown {...props}>{children}</Dropdown>
)

Header.Search = ({ searchTerm, setSearchTerm, ...props }) => {
  const [searchActive, setSearchActive] = useState(false)

  return (
    <Search active={searchActive} {...props}>
      <SearchIcon onClick={() => setSearchActive(prev => !prev)}>
        <MdSearch />
      </SearchIcon>
      <SearchInput
        value={searchTerm}
        onChange={e => setSearchTerm(e.target.value)}
        placeholder="Films, series"
        active={searchActive}
      />
    </Search>
  )
}

Header.PlayButton = ({ children, ...props }) => (
  <PlayButton {...props}>
    <IoMdPlay />
    {children}
  </PlayButton>
)

Header.InfoButton = ({ children, ...props }) => (
  <InfoButton {...props}>
    <MdInfoOutline />
    {children}
  </InfoButton>
)
