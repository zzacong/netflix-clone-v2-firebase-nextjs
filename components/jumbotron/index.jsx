import {
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
  Item,
} from './styles/jumbotron'

export default function Jumbotron({ children, direction = 'row', ...props }) {
  return (
    <Item direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  )
}

Jumbotron.Container = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}
Jumbotron.Pane = ({ children, ...props }) => {
  return <Pane {...props}>{children}</Pane>
}
Jumbotron.Title = ({ children, ...props }) => {
  return <Title {...props}>{children}</Title>
}
Jumbotron.SubTitle = ({ children, ...props }) => {
  return <SubTitle {...props}>{children}</SubTitle>
}
Jumbotron.Image = ({ children, ...props }) => {
  return <Image {...props} />
}
