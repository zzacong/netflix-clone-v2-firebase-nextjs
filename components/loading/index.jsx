import { LockBody, ReleaseBody, Spinner, Picture } from './styles/loading'

export default function Loading({ src, ...props }) {
  return (
    <Spinner {...props}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} />
    </Spinner>
  )
}

Loading.ReleaseBody = () => <ReleaseBody />
