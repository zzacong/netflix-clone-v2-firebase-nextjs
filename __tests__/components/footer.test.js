import { render } from '@testing-library/react'
import { Footer } from '../../components'

describe('<Footer />', () => {
  test('renders the <Footer /> with populated data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions? Contact Us</Footer.Title>

        <Footer.Break />

        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
            <Footer.Link href="#">Investor Relations</Footer.Link>
            <Footer.Link href="#">Ways to Watch</Footer.Link>
            <Footer.Link href="#">Coporate Information</Footer.Link>
            <Footer.Link href="#">Netflix Originals</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Help Centre</Footer.Link>
            <Footer.Link href="#">Jobs</Footer.Link>
            <Footer.Link href="#">Terms of Use</Footer.Link>
            <Footer.Link href="#">Contact Us</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Account</Footer.Link>
            <Footer.Link href="#">Redeem Gift Cards</Footer.Link>
            <Footer.Link href="#">Privacy</Footer.Link>
            <Footer.Link href="#">Speed Test</Footer.Link>
          </Footer.Column>

          <Footer.Column>
            <Footer.Link href="#">Media Centre</Footer.Link>
            <Footer.Link href="#">Buy Gift Cards</Footer.Link>
            <Footer.Link href="#">Cookie Preferences</Footer.Link>
            <Footer.Link href="#">Legal Notices</Footer.Link>
          </Footer.Column>
        </Footer.Row>

        <Footer.Break />

        <Footer.Text>Netflix Clone</Footer.Text>
      </Footer>
    )

    expect(getByText('Questions? Contact Us')).toBeTruthy()
    expect(getByText('FAQ')).toBeInTheDocument()
    expect(getByText('Investor Relations')).toBeInTheDocument()
    expect(getByText('Ways to Watch')).toBeInTheDocument()
    expect(getByText('Coporate Information')).toBeInTheDocument()
    expect(getByText('Netflix Originals')).toBeInTheDocument()
    expect(getByText('Help Centre')).toBeInTheDocument()
    expect(getByText('Jobs')).toBeInTheDocument()
    expect(getByText('Terms of Use')).toBeInTheDocument()
    expect(getByText('Contact Us')).toBeInTheDocument()
    expect(getByText('Account')).toBeInTheDocument()
    expect(getByText('Redeem Gift Cards')).toBeInTheDocument()
    expect(getByText('Privacy')).toBeInTheDocument()
    expect(getByText('Speed Test')).toBeInTheDocument()
    expect(getByText('Media Centre')).toBeInTheDocument()
    expect(getByText('Buy Gift Cards')).toBeInTheDocument()
    expect(getByText('Cookie Preferences')).toBeInTheDocument()
    expect(getByText('Legal Notices')).toBeInTheDocument()
    expect(getByText('Netflix Clone')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
