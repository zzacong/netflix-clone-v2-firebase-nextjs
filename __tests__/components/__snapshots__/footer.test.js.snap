// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders the <Footer /> with populated data 1`] = `
<footer
  class="footer__Container-y863vs-0 jxhylM"
>
  <p
    class="footer__Title-y863vs-4 elwnGE"
  >
    Questions? Contact Us
  </p>
  <div
    class="footer__Break-y863vs-6 eavZkT"
  />
  <div
    class="footer__Row-y863vs-2 hwhVlo"
  >
    <section
      class="footer__Column-y863vs-1 guRtpN"
    >
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        FAQ
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Investor Relations
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Ways to Watch
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Coporate Information
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Netflix Originals
      </a>
    </section>
    <section
      class="footer__Column-y863vs-1 guRtpN"
    >
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Help Centre
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Jobs
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Terms of Use
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Contact Us
      </a>
    </section>
    <section
      class="footer__Column-y863vs-1 guRtpN"
    >
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Account
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Redeem Gift Cards
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Privacy
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Speed Test
      </a>
    </section>
    <section
      class="footer__Column-y863vs-1 guRtpN"
    >
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Media Centre
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Buy Gift Cards
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Cookie Preferences
      </a>
      <a
        class="footer__Link-y863vs-3 jvfZGr"
        href="#"
      >
        Legal Notices
      </a>
    </section>
  </div>
  <div
    class="footer__Break-y863vs-6 eavZkT"
  />
  <p
    class="footer__Text-y863vs-5 bLrDSf"
  >
    Netflix Clone
  </p>
</footer>
`;
